src/managed/Microsoft.DotNet.Build.Bundle/BundleManifest.cs

```diff
@@ -0,0 +1,208 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.DotNet.Build.Bundle
+{
+    /// <summary>
+    ///  BundleManifest is a description of the contents of a bundle file.
+    ///  This class handles creation and consumption of bundle-manifests.
+    ///  
+    ///  Here is the description of the Bundle Layout:
+    ///  _______________________________________________
+    ///  AppHost 
+    ///
+    ///
+    /// ----------------------------------------------
+    /// The embedded files including the app, its
+    /// configuration files, dependencies, and 
+    /// possibly the runtime.
+    /// 
+    /// 
+    /// 
+    /// 
+    /// 
+    /// 
+    ///
+    /// -----------------------------------------------
+    /// Meta-data Directory (FileEntries)
+    ///   "app.deps.json"               (if any)
+    ///   "app.runtimeconfig.json"      (if any)
+    ///   "app.runtimeconfig.dev.json"  (if any)
+    ///   "app.dll"                 (The main app)
+    ///   "IL_1.dll"                (Assemblies, if any)
+    ///    ...
+    ///   "IL_n.dll"
+    ///   "File_1"                  (Other Files, if any)
+    ///    ...
+    ///   "File_n" 
+    /// - - - - - - - - - - - - - - - - - - - - - - - -
+    ///  Bundle Manifest (footer)
```

Comment:
Not sure if my comment on this was lost - basically I would restructure this such that there's an offset to the manifest right behind the signature. So that reader validates the signature, then reads the offset and jumps to it to read the manifest.

Without that indirection we're effectively freezing the size of the manifest forever (since reader jumps by a constant instead). Future proofing the format by not having constants feels like a good idea.